streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessor.java

```diff
@@ -28,32 +28,38 @@
 import org.apache.kafka.streams.processor.internals.InternalProcessorContext;
 import org.apache.kafka.streams.processor.internals.ProcessorRecordContext;
 import org.apache.kafka.streams.state.internals.ContextualRecord;
+import org.apache.kafka.streams.state.internals.buffer.InMemoryTimeOrderedKeyValueBufferStore;
+import org.apache.kafka.streams.state.internals.buffer.TimeOrderedKeyValueBuffer;
+
+import java.util.Objects;
 
 import static java.util.Objects.requireNonNull;
 
 public class KTableSuppressProcessor<K, V> implements Processor<K, Change<V>> {
     private final long maxRecords;
     private final long maxBytes;
     private final long suppressDurationMillis;
-    private final TimeOrderedKeyValueBuffer buffer;
     private final TimeDefinition<K> bufferTimeDefinition;
     private final BufferFullStrategy bufferFullStrategy;
     private final boolean shouldSuppressTombstones;
+    private final String storeName;
+    private TimeOrderedKeyValueBuffer buffer;
     private InternalProcessorContext internalProcessorContext;
 
     private Serde<K> keySerde;
-    private Serde<Change<V>> valueSerde;
+    private FullChangeSerde<V> valueSerde;
```

Comment:
These types are almost equivalent, but I realized that it's better to be safe. We need to know that our serde is capable of handling Changes with both sides populated.